massa-bootstrap/src/server_binder.rs

```diff
@@ -104,13 +104,48 @@ impl BootstrapServerBinder {
 
         Ok(())
     }
+    fn read_exact(
+        &mut self,
+        buf: &mut [u8],
+        time_limit: Option<Duration>,
+    ) -> Result<Duration, (usize, std::io::Error)> {
+        let start = Instant::now();
+        let time_limit = if time_limit.is_none() {
+            // self.duplex
+            //     .set_read_timeout(Some(Duration::from_secs(9999999)))
+            //     // Err only if Some(zero-duration) is used
+            //     .unwrap();
+            return self
+                .duplex
+                .read_exact(buf)
+                .map_err(|er| (buf.len(), er))
+                .map(|_| Instant::now().duration_since(start));
+        } else {
+            time_limit.unwrap()
+        };
 
-    pub async fn send_msg(
+        let mut acc = 0;
+        let mut remain = time_limit;
+        let mut clock = Duration::from_secs(0);
+        while acc < buf.len() {
```

Comment:
we can improve a bit that with : 
```rust
        let mut acc = 0;
        while acc < buf.len() {
            if Instant::now().duration_since(start) > time_limit {
                return Err((acc, std::io::ErrorKind::TimedOut.into()));
            }
            // self.duplex
            //     .set_read_timeout(Some(time_limit - clock))
            //     .expect("internal error");
            self.duplex.read(&mut buf[acc..]).map_err(|er| (acc, er))?;
        }
```


that remove `remain`, `clock` and his alloc